librz/main/rz-diff.c

```diff
@@ -1894,6 +1895,9 @@ static void diff_functions_result_as_json(RzAnalysisMatchResult *result, PJ *pj)
 	rz_list_foreach (result->matches, iter, pair) {
 		fcn_a = (RzAnalysisFunction *)pair->pair_a;
 		fcn_b = (RzAnalysisFunction *)pair->pair_b;
+		if (fcn_a->type != RZ_ANALYSIS_FCN_TYPE_FCN && fcn_a->type != RZ_ANALYSIS_FCN_TYPE_SYM) {
```

Comment:
is this logic also in the other output formats?

The reason i usually prefer functions that do:
```
for x in l:
   [logic on x, filtering, whatever...]
   if output == json:
       print json format
   elif output == table:
       print table format
   [...]
```

instead of:
```
if output == json:
   for x in l:
       [some logic on x, filtering, whatever...]
       print json format
elif output == table:
   for x in l:
       [some logic on x, filtering, whatever... potentially different from the ones above...]
       print table format
[...]
```

I think the former is more robust as the logic is the same for all the output modes.

Reply:
yes in all formats.

Reply:
You can setup a separate function then 😜